lib/galaxy/datatypes/binary.py

```diff
@@ -1487,6 +1489,54 @@ def sniff(self, filename):
 Binary.register_sniffable_binary_format("oxli.graphlabels", "oxligl",
                                         OxliGraphLabels)
 
+class Fast5Archive ( CompressedArchive ):
+    """Class describing a FAST5 archive """
+    MetadataElement( name="fast5_count", default='0', param=MetadataParameter, desc="Read Count",
+                     readonly=True, visible=True, no_value=None )
+    file_ext = "fast5_archive"
+
+    def set_meta( self, dataset, overwrite=True, **kwd ):
+        super( Fast5Archive, self ).set_meta( dataset, overwrite=overwrite, **kwd )
+        try:
+            if dataset and tarfile.is_tarfile( dataset.file_name ):
+                temptar = tarfile.open( dataset.file_name, 'r' )
+                contents = temptar.getnames()
+                r = re.compile(".+\.fast5$")
```

Comment:
Can we replace the regex with `.endswith('.fast5')`?
Also filter returns an iterator on python3, so I guess the most efficient way to get the count would be `sum(1 for content in contents if content.endswith('.fast5')`